server/src/main/java/org/apache/iotdb/db/metadata/MLogWriter.java

```diff
@@ -53,40 +58,50 @@ public MLogWriter(String schemaDir, String logFileName) throws IOException {
     logFile = SystemFileFactory.INSTANCE.getFile(schemaDir + File.separator + logFileName);
     FileWriter fileWriter = new FileWriter(logFile, true);
     writer = new BufferedWriter(fileWriter);
+    fileOutputStream = new FileOutputStream(logFile, true);
+    channel = fileOutputStream.getChannel();
   }
 
   public void close() throws IOException {
     writer.close();
+    channel.close();
+    fileOutputStream.close();
   }
 
   public void createTimeseries(CreateTimeSeriesPlan plan, long offset) throws IOException {
-    writer.write(String.format("%s,%s,%s,%s,%s", MetadataOperationType.CREATE_TIMESERIES,
-        plan.getPath().getFullPath(), plan.getDataType().serialize(),
-        plan.getEncoding().serialize(), plan.getCompressor().serialize()));
+    StringBuilder buf = new StringBuilder();
+    buf.append(String.format("%s,%s,%s,%s,%s", MetadataOperationType.CREATE_TIMESERIES,
+            plan.getPath().getFullPath(), plan.getDataType().serialize(),
+            plan.getEncoding().serialize(), plan.getCompressor().serialize()));
 
-    writer.write(",");
+    buf.append(",");
     if (plan.getProps() != null) {
       boolean first = true;
       for (Map.Entry<String, String> entry : plan.getProps().entrySet()) {
         if (first) {
-          writer.write(String.format("%s=%s", entry.getKey(), entry.getValue()));
+          buf.append(String.format("%s=%s", entry.getKey(), entry.getValue()));
           first = false;
         } else {
-          writer.write(String.format("&%s=%s", entry.getKey(), entry.getValue()));
+          buf.append(String.format("&%s=%s", entry.getKey(), entry.getValue()));
         }
       }
     }
 
-    writer.write(",");
+    buf.append(",");
     if (plan.getAlias() != null) {
-      writer.write(plan.getAlias());
+      buf.append(plan.getAlias());
     }
 
-    writer.write(",");
+    buf.append(",");
     if (offset >= 0) {
-      writer.write(String.valueOf(offset));
+      buf.append(offset);
     }
-    newLine();
+    final String newLine = System.getProperty("line.separator");
+    buf.append(newLine);
+
+    channel.write(ByteBuffer.wrap(buf.toString().getBytes()));
+    channel.force(true);
+    ++lineNumber;
   }
 
   public void deleteTimeseries(String path) throws IOException {
```

Comment:
I think you could use `channel.write()` to implement all the functions in this class, including `deleteTimeseries`, `setStorageGroup`, `deleteStorageGroup`, `setTTL`, `changeOffset` and so on